### ISSUE TITLE
"Types of property 'isUninitialized' are incompatible." when extending a useQuery hook

### ISSUE THREAD

**Issue by denchen** [reduxjs/redux-toolkit#2660]
I'm trying to extend a `useQuery()` hook with the newly released `TypedUseQueryHookResult` type introduced in 1.8.4, but I'm getting the following:

```
Types of property 'isUninitialized' are incompatible.
  Type 'true' is not assignable to type 'false'.ts(2322)
```

The code (with [CodeSandbox](https://codesandbox.io/s/react-toolkit-query-types-xw2obb?file=/src/api.ts)):
```typescript
import {
  createApi,
  fetchBaseQuery,
  TypedUseQueryHookResult
} from "@reduxjs/toolkit/query/react";

type MyApiProps = {
  id: string;
};

type MyApiResponse = Array<{
  id: string;
  name: string;
}>;

export const api = createApi({
  reducerPath: "myApi",
  baseQuery: fetchBaseQuery({ baseUrl: "https://some-api.com/api/v1" }),
  tagTypes: ["MyApi"],
  endpoints: (build) => ({
    getData: build.query<MyApiResponse, MyApiProps>({
      query: ({ id }) => `my-api/${id}`
    })
  })
});

export const useGetData = (
  args: MyApiProps
): TypedUseQueryHookResult<
  MyApiResponse,
  MyApiProps,
  ReturnType<typeof fetchBaseQuery>
> =>
  api.useGetDataQuery(args, {
    selectFromResult: (results) => ({
      ...results,
      data: results.data?.filter((d) => d.name === "Harry")
    })
  });
```

Note that this usage of `selectFromResult()` works just fine if I use the auto-generated hook directly in a component.

**Comment by phryneas** [comment]
Weird enough, if you declare the return type like
```ts
export const useGetData = (
  args: MyApiProps
): Omit<
  TypedUseQueryHookResult<
    MyApiResponse,
    MyApiProps,
    ReturnType<typeof fetchBaseQuery>
  >,
  ""
>
```
so essentially omitting a property that does not exist - everything works :thinking:

**Comment by Andarist** [comment]
Most likely `Omit` works because it "downgrades" the union to an object type as `Omit` is not distributive.

I took a quick look with the debugger there, and I think I got good information to help you solve this. If we `Compute` every member of those unions etc then we can reason about this way easier: [TS playground](https://www.typescriptlang.org/play?ts=4.9.0-dev.20220904#code/C4TwDgpgBAShwBVzQLwCgpQD5QN4c0IgCdiB7YgfgC4oBXAOwBMIAzASwYiYG4DCoTAIbAhNeszadufAZlZ0ANh0WLuCdgFsIAZVGaw4xiw5de-QhXYBzTkMUBBYtYDORyaZkXMxCAEc6CBdgAEkmdxNpczkoCGYwMk5gADkhbXFg4k5rbAlIs1k5YKFiYHUtXX1DWmMpAu8oYuA6F1oARUDiED0RFoA6Rk52YHZ7dgAvLxiAYzpSOOAAERExWgBZEAcwdjgXBIYXaBxaz2i5dhcAGTIhJmzaVntDwoELgDF4aYALe6hHxWeDQuOjo02mQVafyeEBehAuAFFSBQHtDYZgLgBVBhDEZjSZMWjAYiBNFQXysT5fWgACgAlFAUAA+KAAN0SZ0wAF8CDh8DESOQqLQPsBvgAhISHDokECIwW5HQkUaKCbcOUUXInKKkhTKdiqcraHoGcQMOiaABGJE1Hm1DSstgY9icrnEGy27AACuQwC4bfkpnJfAEgqFwrRMtl-XVAwI4kwEklUukI0So8dbfUYsVSobKmlqlAzZbrRmAxzCE0Wu1Ot1RM0XKTZvMGEsVm7NttdvtDtHTqTMdjsbiVfiUQCYUCXCCwRDx4CYgikcR55PF1cbncGNZCcS1+cXCLvr8LWQyGohAxScJRDVM7HCOTKTT6UzWezYdzMLz7VlHc7nDcdZOy9H0-TLGMKx8fxAmCMIMjTbc+ztfl4kSVtkwgBCsiQiD+waHMyiYDQjSqU1zStYhkKzIp62rKBpS6HoGybOZfFbZZRA7D1uzIA4jjySCBxcLEcWVVUCShCdhOuW5fn+BcD3VFcpMU15p1BcEXEhIkSSnI8fm3Xc9P5Zc73La8VmAnigh7fcBF1FQ1GIipjTAWhi0o0kn1FKkoDpBlmTZdgKy-XI+TkB07EcQDuO2b0yF9aiH2gkM4PDRpEJyPCULkeNEwwtIsNTHDssE-Ds1EXMXNIgtyJLKicpogQq0hRi616Rsp1E4dxLHVT7LhDc5KMgbhOU1dhJnLSdL3YSDPk1EGgFZFytygQbyEayu1svi1MIRz9Wckj8wMDyKJIViWzbW8oHdHa9j2wboIpXyX0C98Qs-Hk8AaTa4p2Xb+OSqC-iUJy8zc+rKJB0koqdGLXW20DEvAtbmsfGDQ3gkr03RlLYjQpMiuwvGtQxzBCMhsjzoa2GCLotra2YlorvYm7VjukDeOBpqCcHMS8W4Sap1krcdzG-TKUW6Sp2mudJfXCaoF057YjMqADIlKVa2UhUlSFphlO8thn3818go-CxOT4NBQEgKB4QADzAYgkAd9Bv1+mJNtoXAoBC3Htx4IsiqDnJOQAbQAXXp0zBQiISGkOg0atOwtyYJ+GAKR-HQeDWCw0Tiq8qJwqU0y0q49okoiJOqHzKTyrOprGUWZcAYh2GPqCebdnOM5+7Ace3m85kzcZf29FD2l0aFLV4FNIV+fxo1lfuq7kcJOMtWfO+d632C0KfoijarLwAPJMjYPQ+0cOoCj2O+dBlahU1yltYgdq9ZwRUskNn+Y9k7gyOtTOqtMYbPzhn+aKLogJcw9AlJKUCGgF2xhla+ZVM4vzLikEm98UHN2qvXGmRYLqNSAc3BsrcmKMzZgsAeAMea9kIQeHq3dDYi3XPLbSXClJryWuuMWvxVbzVnhLU854ICXhNq9fe5sPpH2+l7U+hBfYX0DpXbIIcnR3y0UhR+1cBDZ0RvAoeSC0bYJNmlIuBDKGlwTOhPBFdMFGMrFVOurlSGeVLPYlqjMaEdRYg0PuDD2zI2YQJKxG9BajmForA8wi56CP4YKPh6keGzRMgeBao1RHLQ1tEmIKdjpeILBAy6qDTZvQUYfK2Agwo-h9uff2mjME6LDvoiOMc3GYBMXAgGFjelkixulUmuE-FEFwZhcZWD7ygypmnBuZC6asP8S3BizM6ExN6pwhJ6kkkS3XkrARstuFL14SrOaUtfKTzVq-RuJcHIgNTiQ8pKyvIhLYmE26Q9ImyLNgFOpX1rYnz+i0y+4cOl6NcYYtZB0XmlNqiaCpFCimRRgQjAZyMhnwtSoXHGXThn5ScTMux6L1nELKSij5viKXuI2e1du9COLhIQQ9OywyBa7LiZJY5iSJ7JLOTk8R6ShqZLFdPZW+T46rS1pKL+utlz63-ry42VS5F+SBZbEFDTbbTD2sAKA3wIDTAANa0DgIgZADI8CcigJKR2Ls3bICAA). You can see that the reported error is essentially the same here. 

We can also sort all keys everywhere to make comparisons easier, but that changes the reported error (which is fine, it's just order-dependant): [TS playground](https://www.typescriptlang.org/play?ts=4.9.0-dev.20220904#code/C4TwDgpgBAShwBVzQLwCgpQD5QN4c0IGMBXAJzIgDtgARAQ2HoH4AuKAWRAEEwBLOAGcwAeyqDoOElQAmEAGZ8qEGQG4ChKDMYt20uYuVqNhajNFLgAOXoBbCGyiDgZJQHNsUfQqUr1mzAgKETJHb0M-E0x5EgAbRVjYlQQ+ewBlJlswMNkfI38AvkEAUWCydnl6WIkCzSKAMXgiAAt3CqqaqKgigBkRehk2qErqiFrCIrSSIiIIQUF20fHMIoBVKiU+YD4qvgAvFXYXEjGukL43JSruMjdBHINfYwCoSnkm5vYACgBKKBQAHxQABuIj4zwClAAjidnABJGQPPKRF7OehkYDJVIQDJ2bJ6XIRCGaNHAEgLKAARROZBAuLJggAdNJNttdgdiVAAL4EHD4F6kCjUOg6RxcXgCOaicSSLyEp7LLQ6AmPfJdMwWGg2eyOZyuKgeKTytUvIJkEKORrAFoAIXoEmpQRApXNZE8aSCO1i+xULpCnnCCq6MXifESWPSmXxUCoJFsACMggHjSjCiUyotOi8Gh8hvGRCIkvQqIrev1BgajmQTqXBFMZnMKSMs2n1qyvT6ZJnTi9zpcqNdbvd2OL+AAFc1gQTJ1WpzRvD7fP6AkFgzmYaGw4AI3Uudwz5HrpxMDERnFRxyxhNJo2zo+k8nsR20+nk8Y8zB8rqCyg0BhMMUeH4IRpQkA8iUVbQmBHIDJWEMQW00DUwS1OwHHYPV91vQ9FTNEIVRw4M4gSJIZBSSM8XYK9EzIWs-XKYYOh7NMrRaIZjmYupBD6AYhmbTiJjraZZnmKsay6NYNg2NlvQ5btFT7K5YhuO5AIlCcRCncCgxeBdrU+KBfn+IFQXBRVNzmbdEQwvcDW0k0AjRU8yOxXEskvOMaPsudCAfClnzpJgGXfXk8G-chfxFaDOFgkCEIEzAoPoGCJTimVcNkTVrDQ3d9UNOU71wjMCsIl4QxIs83LAKjPKCOjiv42tWNaStGKWCTuPLPimNresRIpDjazbaSOzktrEMIRSB2Uoc1PHSdp2wiCuj0lol2M1czJWiAYUsncbLy7z7xPTEXIo9yauvN0lp0xygsfKkaUCxg3xMD9PH5AIf2Ff9dBi1KpXio7INFFLgMBmVgfVTKUOynUDqwkrltNYrmrtB0nvo91PXZX0ykVcqw1I8jzzxDyrqh7N01dMSEu6QRmu69qqZ4is3Hkjq+sbDmqeGrZRsOca6amwdVLBvgNK0m6HPnBRF0M5cTLXcydq3fbj0O6WfMwJzTpJqrya8rXjpe-yntfQR33UNBQEgKBigADzAMgkDt9BPzCgUIp+0H-vB+DIeNkHotwbouw19xVBjNCEbsrkAG0AF1KYCZDLG1dCI7soP1TKJFkYCQnwzO0mLqR26uPonmWNzVrGo61mmYmlYhIbUShaGqT+dx8PBrOVx+1F4dy5lwhVoMoyV1Mo8LPhayR+1490T11yLwIguSXus2nQt5ku5kzsQo9z7NG+v9fdHODQNlQNR8S5U8DD2O3Cjgd7Gf7kk5TpCYfTnKP5zijV0loPjowgAFLGOAPSuB7pAheR4i7E1XmTS6Rt4H1Rph3DqjNWr5kLBAYstZG6tT7lTLm7d66833gLcOlCAgixmmLP2EsFrfzHnLfS60p7K22rtOeuVEa30XrrSqa8Yy1Wuugrofknzmy3kfD64UhTnwAuLNKYFAEBCSuwUO4Jn6vxjlnDwCdk6aJ-uYWGGcBHZykUA-CtjC7ESJqIyi4iroYPsXQriOD2ZQFIWmYhvivGCXIQNasdNJLth7tXTQDCVLD0vpLRaDjZbvE4QrDa08VZ8KstY-KQiTbOX1mI6iN4Um+S3rIne8i3qhRPsQb2KjkrMPUXTbRj89FGIMe-Ixn9THlMCL-VC8NelmNMMVApBMnHF2Ka40ptEOpVywVTHxMTBKBLWS3UJtNO5RNkoLYJmA4mzXFkkthG4OFrQydwraulVZ7XnphGxkzpEnRcWXeZ5yl4MiqS+Gpmh3pfi9soqKf1L6tK+e03R4cnkv2jj02FfSvlp2GZnRFYzAio1AfacBmMyjYxgfsmQ9Epmhhmcgj5EivlFCWf47xtcgk9Qbl1OuTKyHCW5ss1s1DolctiQPJS8S5osM0skl5dy0lXMnkrW5kJ7n8IAQMpeRSKXRk+Ri75D0AoWytmgNARAELACgC0CARAADW7A4CIGQP8PAXIoD2ntk7F2yAgA) 

We can also remove all but first types in the `ExprType` union because as far I've seen in the debugger it's already the first member of that union that doesn't match: [TS playground](https://www.typescriptlang.org/play?ts=4.9.0-dev.20220904#code/C4TwDgpgBAShwBVzQLwCgpQD5QN4c0IGMBXAJzIgDtgARAQ2HoH4AuKAWRAEEwBLOAGcwAeyqDoOElQAmEAGZ8qEGQG4ChKDMYt20uYuVqNhajNFLgAOXoBbCGyiDgZJQHNsUfQqUr1mzAgKETJHb0M-E0x5EgAbRVjYlQQ+ewBlJlswMNkfI38AvkEAUWCydnl6WIkCzSKAMXgiAAt3CqqaqKgigBkRehk2qErqiFrCIrSSIiIIQUF20fHMIoBVKiU+YD4qvgAvFXYXEjGukL43JSruMjdBHINfYwCoSnkm5vYACgBKKBQAHxQABuIj4zwClAAjidnABJGQPPKRF7OehkYDJVIQDJ2bJ6XIRCGaNHAEgLKAARROZBAuLJggAdNJNttdgdiVAAL4EHD4F6kCjUOg6RxcXgCOaicSSLyEp7LLQ6AmPfJdMwWGg2eyOZyuKgeKTytUvIJkEKORrAFoAIXoEmpQRApXNZE8aSCO1i+xULpCnnCCq6MXifESWPSmXxUCoJFsACMggHjSjCiUyotOi8Gh8hvGRCIkvQqIrev1BgajmQTqXBFMZnMKSMs2n1qyvT6ZJnTi9zpcqNdbvd2OL+AAFc1gQTJ1WpzRvD7fP6AkFgzmYaGw4AI3Uudwz5HrpxMDERnFRxyxhNJo2zo+k8nsR20+nk8Y8zB8rqCyg0BhMMUeH4IRpQkA8iUVbQmBHIDJWEMQW00DUwS1OwHHYPV91vQ9FTNEIVRw4M4gSJIZBSSM8XYK9EzIWs-XKYYOh7NMrRaIZjmYupBD6AYhmbTiJjraZZnmKsay6NYNg2NlvQ5btFT7K5YhuO5AIlCcRCncCgxeBdrU+KBfn+IFQXBRVNzmbdEQwvcDW0k0AjRU8yOxXEskvOMaPsudCAfClnzpJgGXfXk8G-chfxFaDOFgkCEIEzAoPoGCJTimVcNkTVrDQ3d9UNOU71wjMCsIl4QxIs83LAKjPKCOjiv42tWNaStGKWCTuPLPimNresRIpDjazbaSOzktrEMIRSB2Uoc1PHSdp2wiCuj0lol2M1czJWiAYUsncbLy7z7xPTEXIo9yauvN0lp0xygsfKkaUCxg3xMD9PH5AIf2Ff9dBi1KpXio7INFFLgMBmVgfVTKUOynUDqwkrltNYrmrtB0nvo91PXZX0ykVcqw1I8jzzxDyrqh7N01dMSEu6QRmu69qqZ4is3Hkjq+sbDmqeGrZRsOca6amwdVLBvgNK0m6HPnBRF0M5cTLXcydq3fbj0O6WfMwJzTpJqrya8rXjpe-yntfQR33UNBQEgKBigADzAMgkDtlAwswb6-1B-7wfgyHjYKJL2Fwbouw19xVBjNCEbsrkAG0AF1KagZDLG1dCI7swOCDw0ICOR4ZiKJyqLwL26inonn6cZ1rGoIMteLrnqG6EhtRKFgpJPbXHw8GggRZmsWkdu1aDKMldTIhCz4WskeHN10uyfLhf7rNp0LeZKT+d79QuWtogEOAKAWggIgAGt2DgRBkH+PAuSge17adl3kHUIA)

Then we can drop the fourth element of the `RetType` because it clearly can't be matched based on the `currentType`: [TS playground](https://www.typescriptlang.org/play?ts=4.9.0-dev.20220904#code/C4TwDgpgBAShwBVzQLwCgpQD5QN4c0IGMBXAJzIgDtgARAQ2HoH4AuKAWRAEEwBLOAGcwAeyqDoOElQAmEAGZ8qEGQG4ChKDMYt20uYuVqNhajNFLgAOXoBbCGyiDgZJQHNsUfQqUr1mzAgKETJHb0M-E0x5EgAbRVjYlQQ+ewBlJlswMNkfI38AvkEAUWCydnl6WIkCzSKAMXgiAAt3CqqaqKgigBkRehk2qErqiFrCIrSSIiIIQUF20fHMIoBVKiU+YD4qvgAvFXYXEjGukL43JSruMjdBHINfYwCoSnkm5vYACgBKKBQAHxQABuIj4zwClAAjidnABJGQPPKRF7OehkYDJVIQDJ2bJ6XIRCGaNHAEgLKAARROZBAuLJggAdNJNttdgdiVAAL4EHD4F6kCjUOg6RxcXgCOaicSSLyEp7LLQ6AmPfJdMwWGg2eyOZyuKgeKTytUvIJkEKORrAFoAIXoEmpQRApXNZE8aSCO1i+xULpCnnCCq6MXifESWPSmXxUCoJFsACMggHjSjCiUyotOi8Gh8hvGRCIkvQqIrev1BgajmQTqXBFMZnMKSMs2n1qyvT6ZJnTi9zpcqNdbvd2OL+AAFc1gQTJ1WpzRvD7fP6AkFgzmYaGw4AI3Uudwz5HrpxMDERnFRxyxhNJo2zo+k8nsR20+nk8Y8zB8rqCyg0BhMMUeH4IRpQkA8iUVbQmBHIDJWEMQW00DUwS1OwHHYPV91vQ9FTNEIVRw4M4gSJIZBSSM8XYK9EzIWs-XKYYOh7NMrRaIZjmYupBD6AYhmbTiJjraZZnmKsay6NYNg2NlvQ5btFT7K5YhuO5AIlCcRCncCgxeBdrU+KBfn+IFQXBRVNzmbdEQwvcDW0k0AjRU8yOxXEskvOMaPsudCAfClnzpJgGXfXk8G-chfxFACYIlECENlQMHM0KDdE4WC4plbyj2QyxtXQ499UNOU71wjMoFY5o7QdGlnTKd1PXZX0ykVEMSLPNzo2om9isI7N01dMSBJWQQKr4pjax4is3HkiShIbUTGKWWa22kjs5MWxDCEUgdlKHNTx0nadsIgro9JaJdjNXMzTogGFLJ3GzCqyxUnMxFyKPcqjPO6xKfMwPynxq19BHfdQ0FASAoGKAAPMAyCQSGUDCzAf2Ff9UtHODQISlMIRS9hcG6LsCvcVQYzQx79y5ABtABdZ6CBy1CdUpuzjp0qA8NCAiTuiYiw1I8jzzxJFee6fr8I25ic30saluGya5ZbSZhMbGbhpWrY1sOKAOIKbbB1UnmObOgyjJXUyIQs+FrJ6sXXvai9jYcgGqSBoLyWZKStca4wuTBogEOAKAWggIgAGt2DgRBkH+PAuSge0odh+HkHUIA)

Then we can drop 2 first members of that `RetType` union too, because `data` property doesn't match: [TS playground](https://www.typescriptlang.org/play?ts=4.9.0-dev.20220904#code/C4TwDgpgBAShwBVzQLwCgpQD5QN4c0IGMBXAJzIgDtgARAQ2HoH4AuKAWRAEEwBLOAGcwAeyqDoOElQAmEAGZ8qEGQG4ChKDMb12XXgIjCxE9ZszUZopcABy9ALYQ2UQcDJKA5tijS5i5TUNQggKETJ2PwUlFTNzeRIAG0VExJUEPicAZSYHMHYqEgcAI1C4zT5BAFEwiKh5ekTTYMxKgDF4IgALL3Z3EghywkqAGRF6GV76xubzKEqskiIiI0E+sgGh1sEAVSolPmA+Rr4ALxV2BqbBlqhwvk8lRu4yT0EXfX4ABTIRMEEfFEArFbpR5J0uuwABQASigKAAfFAAG4iPhBOaUACOAzcAEkZC43B4qN4pLJooEtq4mGRgOlMhAco4wC5CiVQoCKcCMeY3IwSGsoABFAZkEDM4CCoYAXwIOHwc1IFGodB0Hx4-CEonEkl83JivM02iYGoM2pMeqBhuplmsNHsTiJ7i8XP8NtuoV+dQ6wG6ACF6BJRaEQDVvT4sqFjokzipw+FqQlknxUgzsrlWQUiqUyG7KSC5pUE3V+jci4Jfd0pldZuZRuNJqTLjNy-XBItlqsW9dqZU9gcjidzjIe3XNPdHlRnq93npNXwfn8AeT3VTQQoIdC4YiUWijYRsbjgATnSSyfq14W+bT6TIMhmWWyc5zVwWD5h+VKhSHxZLpcEMrqGgoCQFAVQAB5gGQSBgSgeAEMqlA0AwprzuaRg6hI+Y8nEJq6Hg8yjjS56qFA05OOwxKujKADaAC6OEehYsj2nYjjOFRLqkkx64WLULjWnx9RJCkaT3oyzJ5IJBrCcWtRjm27QQjWrZxA2Eyqb2BALEsKyCEKtZKbs+z7EOsYjusmwEJOTyJC8bwyVevJgluUCwvCSKouicRHkYJ6EpEsnXvydLpkymZOe+cRfoK7C-hKTDfgAdNIg4xnGQRAWgaBECYwBQN0EBEAA1uwcCIMg8J4DKUBBuBUEwcg6hAA)

And now we can conclude that those are not available at least because the error property is required in one of those union members of the `TypedUseQueryHookResult` (even though it's value is `undefined`) but it's optional in the `UseQueryHookResult`. This is defined on the **third** union member of the `TypedUseQueryHookResult` but the order in the union is not guaranteed so perhaps we shouldn't rely on this - it's likely that this matches the definition order though.

I think I've chosen a slight detour here, in this reasoning, because the easiest would be to just name all of the union members, like [here](https://www.typescriptlang.org/play?ts=4.9.0-dev.20220904#code/C4TwDgpgBAShwEYoF4oG8BQUoQE64HtcB+ALigFcA7AEwgDMBLKiGgbiyhoENhuzKtBs1Yds9CgBsmkyawAqjALYQAynyVgB1OkxbtORRgHNm3SQEFcxgM7ahe0Z1wQAjhQg3gASRr3dIgbYELRgBMzAAHLcKgJeuMzGUAA+ggH6YlBe3LjACspqGlrkOsIZnNnAFDbkAIoeuCDqvNUAdNTMjMCM5owAXk7YAMYU+CHAACK8-OQAsiAWYIxwNmFUNtCppY5BUIw2ADIE3DSJ5PTmG5n7AGLwQwAWZ1AXklec+6oUQ0OeNS+XCDXGwAUXwRHOgOBAFUqJ1ur0BjRyMBcB5Mi56PcHuQABQAShQAD4oAA3cIGAC+HAwoEgsHgACYUOhOHhCCRyHdgI8AELcDb1PAgMEclJQVR4HqSfqsUVEcXbQKZCTSRiyfIqZqaARUChKABGeEVDmVhgSpio5istgE80WjAACoQwDYTelBlAXO5PD4-OR4ol3WVPSEaGEItFYgHUUGtqbythsrlNYUYsUoHrDcb4x7dpVqnUGk0+FUbJkRmMqJNpnaFksVmsNsGdjC4XCETKkZC3kCPjYvj8-j33th9vLcCO+2PDsdTlRjCi0dO9jZuY9ngaCAQ5NwqJkeHwSgnPZjsXjCcgSeTGFSaXToHBgABmFmYbBGS3W6x2Ob1p0um6uYhrs3oeF4vhxLGC4tmawShOE1ZRhAUEJDBwGthUfApjQihakUur6kauCwYmWSloWUBCo0zRlhWowuNWUx8HWDqNgQ6ybGkIFtvC0qysiAK9sCRwnM8ryjquE5TsCg6-DY-youi-brk8C5LspwTgpO3GYdgh7cH+bGeE2K6qjIci4QU2pgOQWbERiDDnlABLEmSFIcNSGC0uAj7wAALG+5omGYlg-qxSzOgQrqkaebjgX6qFxrpcE4AhkYxChMZoUkGGpcmeRWfh6aEdmJF5WRBb-NRJYtOW-awnxiKsDJ-aifOi5CZJ47aa1M5ycOUBKSutzYuJUJsr1KVkQZRkNiZHGSeZ6qWXhaaaHZRF4PRVY1keUD2vNqyLSuZ48jiLmXteHkYF5Pn0k+ACsQX6bWc3LAtnGxbsy0akV60ZvZObTZ6n6hTav4Hf+UUxRVcU+hB-rkTl32ZGGEZIZlSXoSD+bYYVa02aVxGo1hdVFsKtHVDtjF7TMUPGcdX1w7s+yNR2-Hdl1I2zmJ6nc8CqnjcJ-YDQpfVSVNw1o1Nqn8oKxYTuKkoJM1NATo5WLnRebk3ne3kPgywDyL5LJPkgqRPsylvwK+NvAIF9uPfepsggAHmAuAm-SqDvlw0zkGgeyCYGC5sJmmXZUGlIANoALqk1pHL+DxnC-at1kEceeaZGDVphba2ep9gYG+pBRd6Wl4aIVEWNRzjSqVfjqZExX+UUdVxZUzY7Ttl0nOepWtPMfTh0fUzzYsyJc7C91a5jfzEk82L-xL8C0kCw1fedgJGmnU52uXbrN13UMi3AFAjwQEMADW5BPt70C+5SUAClA7ue4-HBAA). And then to compare them one by one with the `ExprType` - this allows us quickly to check if the reported error matches what we were seeing in the app~ or not. Based on that we can see that `isUninitialized: true;` is only in one of the union members of the `TypedUseQueryHookResult` so it's the only potential candidate from this PoV BUT at the same time its `data` property doesn't match the property type in the slimmed down `UseQueryHookResult` (we can't assign something to what expect `undefined`).

TLDR: there are a couple of assignability problems here.
